config/config.go

```diff
@@ -430,7 +432,12 @@ func (cfg *Config) NewNode() (host.Host, error) {
 	}
 
 	if err := cfg.addAutoNAT(bh); err != nil {
-		rh.Close()
+		app.Stop(context.Background())
+		if cfg.Routing != nil {
```

Comment:
This should be part of the stop lifecycle, no? Maybe we defer the fix until later

Reply:
wdym? We are not closing the host with the Stop lifecycle hook.